### ISSUE TITLE
Multi-file detection for `_preamble.tex` and `_postamble.tex` files generated by Tectonic

### ISSUE THREAD

**Issue by parksb** [lervag/vimtex#2490]
### Description

Hi, I'm using Tectonic to compile documents by configuring `let g:vimtex_view_method = 'tectonic'`.

Tectonic recommends `tectonic -X new sample` command which generates `_preamble.tex`, `_postamble.tex`, and `index.tex` files under the `sample/src` directory to create a new TeX project. Each file contains:

```tex
% _preamble.tex
\documentclass{article}
\title{My Title}
\begin{document}
```

```tex
% index.tex
Hello, world.
```

```tex
% _postamble.tex
\end{document}
```

When I run `<localleader>ll` in the `index.tex` to compile, VimTex shows an error message. I found that VimTeX works if the contents of `_preamble.tex` and `_postamble.tex` are combined into `index.tex`, but I wonder if is there a way to use VimTeX without any modification to the directory generated by Tectonic.

### Steps to reproduce

1. `tectonic -X new sample`
2. `nvim sample/src/index.tex`
3. Run `<localleader>ll`.

### Expected behavior

VimTeX starts compilation.

### Actual behavior

VimTeX shows an error message:

```
VimTeX: Compilation error due to failed mainfile detection!
        Please ensure that VimTeX can locate the proper main .tex file.
        Read ":help vimtex-multi-file" for more info
```

### Do you use a latexmkrc file?

No

### VimtexInfo

```yaml
System info:
  OS: macOS 12.4 (21F2081)
  Vim version: NVIM v0.7.2
  Has clientserver: true
  Servername: /var/folders/c9/kkczqw7d28bf31_wkkzmrhpr0000gn/T/nvimxiWFtQ/0

VimTeX project: index
  base: index.tex
  root: /Users/my/sample/src
  tex: /Users/my/sample/src/index.tex
  main parser: fallback current file
  document class: 
  compiler: tectonic
    options:
      --keep-logs
      --synctex
  viewer: General
  qf method: LaTeX logfile
```
```

**Comment by lervag** [comment]
Huh, that's strange. The sample generated by Tectonic does not compile with any other tool. This makes me quite skeptical of adding any support for it to VimTeX. In fact, it seems to entirely change the idea of how a project should be structured and how to compile it, and the layout is not compatible with the core assumptions of VimTeX. That is, that the project you are working on can be compiled with any standard compilers and build tools.

So, I'm sorry to say I think this will go under the category "won't fix". :\

You should also notice that a lot of the VimTeX code relies on parsing a project from the main file, and if we can't discover the main file, then this parsing fails. This will prevent things like citation completion and similar from working. :\

**Comment by kunzaatko** [comment]
Just some more information on this:
> You should also notice that a lot of the VimTeX code relies on parsing a project from the main file, and if we can't discover the main file, then this parsing fails.

The main file in a `tectonic` project is always stored in the configuration file of the project root `Tectonic.toml`, so potentially that would be the way to get acquire the main file.

1. Compilation with `tectonic` in a project does not rely on providing the main file and determines the structure from the configuration
2. This structure is based on the new [_V2_](https://tectonic-typesetting.github.io/book/latest/ref/v2cli.html) type interface that tectonic provides and there is at this moment a way to use the old V1 interface. (but the default structure from tectonic would be broken)
3. As already mentioned this is V2 and it will become the default for `tectonic` eventually and the V1 interface will be removed as can be seen [here](https://tectonic-typesetting.github.io/book/latest/ref/v2cli.html). At that time, the `tectonic` value for the `vimtex_compiler_method` will no longer be usefull on the latest version of `tectonic` and should be then mentioned in the docs somewhere.
4. It should not be hard to implement the `tectonic` compiler through the _vimtex-compiler-generic_ since it only requires the command `tectonic -X build` to be called from anywhere within the project. I will be doing this when the generic compiler is documented and will share the configuration here (if it indeed will be possible) #2021.

**Comment by kunzaatko** [comment]
One more thing... You can run `tectonic -X watch` and the project files are watched and compiled continually on changes. With a viewer that supports updating (like `zathura`) it should be an alternative workflow option instead of using the built in continual compilation in `vimtex`... This means that only forward synctex and view is broken. backward synctex should be possible to set up using the zathura configuration.

**Comment by lervag** [comment]
> Just some more information on this: …

Thanks! Interesting and useful! Notice it could be useful with an update of the VimTeX docs for Tectonic, especially the contents under `:help vimtex-compiler-tectonic`. Feel free to suggest an improved text here, or to point out things that may be currently outdated.

> 3. As already mentioned this is V2 and it will become the default for `tectonic` eventually and the V1 interface will be removed as can be seen [here](https://tectonic-typesetting.github.io/book/latest/ref/v2cli.html). At that time, the `tectonic` value for the `vimtex_compiler_method` will no longer be usefull on the latest version of `tectonic` and should be then mentioned in the docs somewhere or perhaps removed entirely?

Agreed. I would be happy to get some help on making this update. Preferably as a PR, but some simple suggestions in an issue thread is also useful.

> 4. It should not be hard to implement the `tectonic` compiler through the _vimtex-compiler-generic_ since it only requires the command `tectonic -X build` to be called from anywhere within the project. I will be doing this when the generic compiler is documented and will share the configuration here (if it indeed will be possible) [Improve generic compiler backend #2021](https://github.com/lervag/vimtex/issues/2021).

I've resolved #2021 now. Let me know what you think.

> One more thing... You can run `tectonic -X watch` and the project files are watched and compiled continually on changes. With a viewer that supports updating (like `zathura`) it should be an alternative workflow option instead of using the built in continual compilation in `vimtex`... This means that only forward synctex and view is broken. backward synctex should be possible to set up using the zathura configuration.

Cool. Would there be any benefits to the user of this compared to using VimTeX for the viewer integration?
